netlify/functions/serverless/cleanse-query-parameters.js

```diff
@@ -67,6 +75,14 @@ module.exports = cleanseQueryParameters;
 
 // VALIDATORS
 
+/**
+ * @param {string} enumName - name of the enum the value should be a variant of
+ * @returns {Validator} - function used to determine whether a specific query parameter should be kept or thrown out
+ */
+function getIsVariantOf(enumName) {
+	return (value) => Object.keys(ENUMS[enumName]).includes(value);
+}
+
```

Comment:
This is neat stuff! I've been considering an alternate that — I feel — could lead to a bit more inlineability. I'm curious to know what your thoughts are:

```js
function oneOf(...args) {
	return (value) => args.includes(value);
}
```

That way, in the `VALID_PARAMETERS` object, you can say:

```js
{
	DEMO: {
		validate: oneOf(true, false, 'STATIC')
	}
}
```

What you'd lose is using the same interface for `getVariantValue`.

Reply:
They serve exactly the same purpose, so I'm fine with both.
I'll refactor so it can be
```js
{
    DEMO: {
        validate: oneOf("true", "false", 'static')
    }
}
```
(Quoted because the validate function checks the value of a query parameter, and that's always a string)

The `getVariantValue` right now isn't _super_ useful because the keys of the enum (what users provide as query parameter) are very similar to the value for that key (what the JavaScript uses internally).
I think it's still useful to be in place, because a query parameter is always a string, and that configuration might lead to using a certain function in the JavaScript. (in this case the string `"false"` as value for a queryparam maps to the boolean `false` to be used in the code.)